amgut/lib/data_access/ag_data_access.py

```diff
@@ -323,8 +340,11 @@ def deleteAGParticipantSurvey(self, ag_login_id, participant_name):
                 self.deleteSample(info['barcode'], ag_login_id)
 
             # Delete last due to foreign keys
-            sql = "DELETE FROM ag_kit_barcodes WHERE survey_id IN %s"
+            sql = """DELETE FROM ag.source_barcodes_surveys
+                     WHERE survey_id IN %s"""
             TRN.add(sql, [tuple(survey_ids)])
+            sql = "DELETE FROM ag.ag_kit_barcodes WHERE barcode IN %s"
```

Comment:
Make sense. I changed the SQL by adding the condition that ag_kit_barcodes only get deleted if there is only one associated source_id left.